frontend/src/app/forum/forum.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ForumChatService } from '../forum-chat.service';
+
+@Component({
+  selector: 'app-forum',
+  templateUrl: './forum.component.html',
+  styleUrls: ['./forum.component.sass'],
+})
+export class ForumComponent implements OnInit {
+  constructor(private ForumChatService: ForumChatService) {}
+  goToPost = false;
+
+  all_posts = [];
+  seenPost = null;
+
+  newQuestion = new FormControl();
+
+  ngOnInit(): void {
+    this.all_posts = this.ForumChatService.getAllPosts();
+  }
+
+  seePost(post): void {
+    if (post == 'back') {
+      console.log('seePost');
+      this.goToPost = false;
+      this.seenPost = null;
+    } else {
+      console.log('seePost');
+      this.goToPost = true;
+      this.seenPost = post;
+    }
+  }
+  vote(post, v): void {
+    console.log(post, v);
+    if (!post.voted) {
+      post.upvote = v == 'up' ? post.upvote + 1 : post.upvote - 1;
+      post.voted = true;
+      console.log('voted' + v);
+      //PUSH TO BACKEND ?
+    } else {
+      console.log('already voted');
+    }
+
+    this.ForumChatService.updatePosts(this.all_posts);
```

Comment:
Maybe we should not always update all posts but just that specific one. Just with the backend implementation in mind. But works for now.